frontend/public/components/routes/create-route.tsx

```diff
@@ -196,11 +227,54 @@ export class CreateRoute extends React.Component<null, CreateRouteState> {
       }));
   }
 
+  newAltServiceEntry() {
+    return {
+      entry: {
+        service: null,
+        weight: 100,
+      },
+      uid: _.uniqueId(),
+    };
+  }
+
+  addAltServiceEntry(){
+    this.setState({
+      alternateServices: _.concat(this.state.alternateServices, this.newAltServiceEntry()),
+    });
+  }
+
+  removeAltServiceEntry(alternateServiceID){
+    const updatedServiceEntriesArray = [...this.state.alternateServices];
+    updatedServiceEntriesArray.splice(alternateServiceID, 1);
+    this.setState({
+      alternateServices: updatedServiceEntriesArray,
+    });
+  }
+
+  onDataChanged(updatedEntry, entryID) {
+    const updatedServiceEntriesArray = [...this.state.alternateServices];
+    const updatedEntryData = {
+      uid: updatedServiceEntriesArray[entryID].uid,
+      entry: updatedEntry,
+    };
+    updatedServiceEntriesArray[entryID] = updatedEntryData;
+    this.setState({
+      alternateServices: updatedServiceEntriesArray,
+    });
+  }
+
   render() {
     const title = 'Create Route';
-    const { loaded, services, service, portOptions, targetPort, termination } = this.state;
+    const { loaded, services, service, portOptions, targetPort, termination, alternateServices } = this.state;
     const serviceOptions = {};
     _.each(_.sortBy(services, 'metadata.name'), ({ metadata: { name } }) => serviceOptions[name] = <ResourceName kind="Service" name={name} />);
+    const configuredServices = _.isEmpty(service) ? [] : [_.get(service, 'metadata.name')];
```

Comment:
No need to use `_.get` here and it loses some type checking

```suggestion
    const configuredServices = _.isEmpty(service) ? [] : [service.metadata.name];
```